Feat(web-react): Introduce ScrollView component #DS-464 (#849)
lmc-eu/spirit-design-system: packages/web-react/src/components/ScrollView/README.md

# ScrollView

## Usage

> ### Default (Horizontal)

Reviewer: That's not true. Use set vertical as default. 
```suggestion
### Default (Vertical)
```

Author: fixed

We need to set the maximum height if it is not defined by the parent.

```javascript

> <div style={{ height: '250px' }}>

Reviewer: I think it is mainly confusing because there is a wrong title here. Also, we don't need a wrapper. Because there are several ways to limit the height, it could be fixed height like this or it could be in some kind of grid layout etc. This should be explained that we set inline height only to showcase the ScrollView.

> <div style={{ height: '250px' }}>

Reviewer: With small help from our AI colleagues
> It's crucial to understand that the ScrollView component, like any HTML element, expands to fit its content. Without a height limit on the parent container, the ScrollView will stretch to include all its children, which could negate the need for scrolling.

> To make scrolling and scroll indicators function correctly, the parent container's height must be limited. In our examples, we set this height limit using inline styles for demonstration purposes only.

> <div style={{ height: '250px' }}>

Author: done in README

  <ScrollView>
    <p>
      Lorem ipsum dolor sit amet, consectetuer adipiscing elit. Aenean commodo ligula eget dolor. Aenean massa. Cum
      sociis natoque penatibus et magnis dis parturient montes, nascetur ridiculus mus. Donec quam felis, ultricies nec,
      pellentesque eu, pretium quis, sem. Nulla consequat massa quis enim. Donec pede justo, fringilla vel, aliquet nec,
      vulputate eget, arcu. In enim justo, rhoncus ut, imperdiet a, venenatis vitae, justo. Nullam dictum felis eu pede
      mollis pretium. Integer tincidunt. Cras dapibus. Vivamus elementum semper nisi. Aenean vulputate eleifend tellus.
      Aenean leo ligula, porttitor eu, consequat vitae, eleifend ac, enim. Aliquam lorem ante, dapibus in, viverra quis,
      feugiat a, tellus. Phasellus viverra nulla ut metus varius laoreet. Quisque rutrum. Aenean imperdiet.
    </p>
  </ScrollView>
</div>
```

> ### Vertical

Reviewer: ```suggestion
### Horizontal
```

Author: fixed

```javascript

> <ScrollView>

Reviewer: ```suggestion
<ScrollView direction="horizontal">
```

Author: fixed

  <p className="py-700" style={{ whiteSpace: 'nowrap' }}>
    Lorem ipsum dolor sit amet, consectetuer adipiscing elit. Aenean commodo ligula eget dolor. Aenean massa. Cum sociis
    natoque penatibus et magnis dis parturient montes, nascetur ridiculus mus. Donec quam felis, ultricies nec,
    pellentesque eu, pretium quis, sem. Nulla consequat massa quis enim. Donec pede justo, fringilla vel, aliquet nec,
    vulputate eget, arcu. In enim justo, rhoncus ut, imperdiet a, venenatis vitae, justo. Nullam dictum felis eu pede
    mollis pretium. Integer tincidunt. Cras dapibus. Vivamus elementum semper nisi. Aenean vulputate eleifend tellus.
    Aenean leo ligula, porttitor eu, consequat vitae, eleifend ac, enim. Aliquam lorem ante, dapibus in, viverra quis,
    feugiat a, tellus. Phasellus viverra nulla ut metus varius laoreet. Quisque rutrum. Aenean imperdiet.
  </p>
</ScrollView>
```

> ## Accordion Props

Reviewer: ? :)

> | Prop name          | Type                       | Default      | Required | Description                 |

Reviewer: You are missing the isScrollbarDisabled prop. Also in the implementation.

Author: done

| ------------------ | -------------------------- | ------------ | -------- | --------------------------- |
| `children`         | `ReactNode`                | -            | yes      | ScrollView children's nodes |

> | `direction`        | `'horizontal', 'vertical'` | `'vertical'` | no       | Direction of the wrapper    |

Reviewer: ```suggestion
| `direction`        | `horizontal`, `vertical` | `vertical` | no       | Direction of the scroll |
```
